airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/state/MongoDbStreamState.java

```diff
@@ -4,4 +4,11 @@
 
 package io.airbyte.integrations.source.mongodb.state;
 
-public record MongoDbStreamState(String id, InitialSnapshotStatus status, IdType idType) {}
+public record MongoDbStreamState(String id, InitialSnapshotStatus status, IdType idType, Byte binarySubType) {
+
+  public MongoDbStreamState(String id, InitialSnapshotStatus status, IdType idType) {
+    this(id, status, idType, (byte) 0);
```

Comment:
Will this allow for existing saved saved binary _id to transition in a non breaking way?
assuming this is subtype 0 (= generic binary).

What would happen to saved UUID (type 4)? 
Do we ensure they will also have a smooth transition?

Reply:
not really - if idType is not binary, dev is not required to provide subType; that field will not be used.

existing state will be going through composeStateFromlastId to find the subtype.